### PR TITLE
Add ability for custom user-agent parsers to be used and add a custom parser for in-app browsers

### PR DESCRIPTION
**Pull request by JakeChampion**
Add custom parser for returning all iOS in-app browsers as Mobile Safari

### CODE REVIEW

**Comment by chee** [comment]
This is wonderful!

 So this works for every iOS in-app browser other than Snapchat?

**Comment by JakeChampion** [comment]
@chee It works for SnapChat also :-)

File: test/user_agent_strings.yaml [hunk 1]
```diff
@@ -0,0 +1,200 @@
+# These are user-agent strings which have been overriden by our parser from what the uap-core project expect them to be parsed as
+test_cases:
+
+  - user_agent_string: 'Mozilla/5.0 (iPhone; CPU iPhone OS 10_2_1 like Mac OS X) AppleWebKit/602.4.6 (KHTML, like Gecko) Snapchat/10.44.1.1 (iPhone7,2; iOS 10.2.1; gzip)'
```
Comment (JakeChampion) [inline test/user_agent_strings.yaml]: Here's a snapchat useragent being tested @chee

Comment (chee) [reply test/user_agent_strings.yaml]: oh nice!!

Comment (chee) [reply test/user_agent_strings.yaml]: this will make so many people happy

File: test/ua.test.js [hunk 2]
```diff
@@ -24,37 +25,32 @@ function fixFixture(f, props) {
...
+const fixtures = _.uniqBy([
+		require.resolve("./user_agent_strings.yaml"),
+		require.resolve("../uap-core/test_resources/firefox_user_agent_strings.yaml"),
+		require.resolve("../uap-core/tests/test_ua.yaml"),
+		require.resolve("../uap-core/test_resources/pgts_browser_list.yaml"),
+		require.resolve("../uap-core/test_resources/opera_mini_user_agent_strings.yaml"),
+		require.resolve("../uap-core/test_resources/podcasting_user_agent_strings.yaml")
+	]
+	.reduce((acc, filename) => acc.concat(readYAML(filename).test_cases), [])
+	.map(f => fixFixture(f, ["major", "minor", "patch"])), 'user_agent_string');
```
[the remark this answers is not in the document]
Comment (JakeChampion) [reply test/ua.test.js]: Yes, this would be a good thing to add :-)

File: package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
   "license": "MIT",
   "devDependencies": {
     "js-yaml": "^3.13.1",
+    "lodash": "^4.17.11",
```
[the remark this answers is not in the document]
Comment (JakeChampion) [reply package.json]: Not sure on this one, what do you think? :)

File: lib/ua_parser.vcl [hunk 4]
```diff
@@ -8,7 +8,12 @@ sub useragent_parser {
   declare local var.Patch STRING;
   set var.Patch = "";
   if (!req.http.User-Agent) {
```
[the remark this answers is not in the document]
Comment (JakeChampion) [reply lib/ua_parser.vcl]: ^_^

**Pull request merged**